app/controllers/herbarium_records_controller.rb

```diff
@@ -319,42 +343,12 @@ def can_add_record_to_herbarium?
     false
   end
 
-  def herbarium_label_free?
-    @other_record = HerbariumRecord.where(
-      herbarium: @herbarium_record.herbarium,
-      accession_number: @herbarium_record.accession_number
-    ).first
-    !@other_record || @other_record == @herbarium_record
-  end
-
-  def figure_out_where_to_go_back_to
-    @back = params[:back].to_s
-    @back_object = nil
-    if @back == "show"
-      @back_object = @herbarium_record
-    elsif @back != "index"
-      @back_object = Observation.safe_find(@back)
-      return if @back_object
-
-      @back_object = if @herbarium_record.observations.count == 1
-                       @herbarium_record.observations.first
-                     else
-                       @herbarium_record
-                     end
-    end
-  end
-
   def redirect_to_observation_or_herbarium_record
     if @back_object
       redirect_to(helpers.object_path(@back_object, q: get_query_param))
     else
-      # redirect_with_query(
-      #   action: :index_herbarium_record,
-      #   id: @herbarium_record.id
-      # )
-      redirect_to herbarium_record_index_herbarium_record_path(
-        q: get_query_param
-      )
+      redirect_with_query(action: :index_herbarium_record,
+                          id: @herbarium_record.id)
```

Comment:
I'd like to use the path helper `herbarium_records_index_herbarium_record` but that doesn't do the right thing with `:id`:
```ruby
(byebug) herbarium_records_index_herbarium_record_path
"/herbarium_records/index_herbarium_record"
(byebug) herbarium_records_index_herbarium_record_path(123)
"/herbarium_records/index_herbarium_record.123"
(byebug) herbarium_records_index_herbarium_record_path(id: 123)
"/herbarium_records/index_herbarium_record?id=123"
```
But I want 
```ruby
/herbarium_records/index_herbarium_record/123
```
Something needs to be added/fixed in the routes. I included that in a follow-up story in order to avoid delaying merger of the changes in this PR.

Reply:
Noted. My rake routes shows that herbarium_records_index_herbarium_record_path should be able to take an :id, but byebug probably isn't lying to you. Hmm. 

Emailing you about it separately.